src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeProvider.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Resources;
+using Bicep.Core.TypeSystem.Types;
+
+namespace Bicep.Core.TypeSystem.Providers.ThirdParty
+{
+    public class ThirdPartyResourceTypeProvider : ResourceTypeProviderBase, IResourceTypeProvider
+    {
+        public static readonly TypeSymbol Tags = new ObjectType(nameof(Tags), TypeSymbolValidationFlags.Default, Enumerable.Empty<TypeProperty>(), LanguageConstants.String, TypePropertyFlags.None);
+
+        private readonly ThirdPartyResourceTypeLoader resourceTypeLoader;
+        private readonly ResourceTypeCache definedTypeCache;
+        private readonly ResourceTypeCache generatedTypeCache;
+
+        public ThirdPartyResourceTypeProvider(ThirdPartyResourceTypeLoader resourceTypeLoader, string providerVersion)
+            : base(resourceTypeLoader.GetAvailableTypes().ToImmutableHashSet())
+        {
+            Version = providerVersion;
+            this.resourceTypeLoader = resourceTypeLoader;
+            definedTypeCache = new ResourceTypeCache();
+            generatedTypeCache = new ResourceTypeCache();
+        }
+
+        private static ResourceTypeComponents SetBicepResourceProperties(ResourceTypeComponents resourceType, ResourceTypeGenerationFlags flags)
+        {
+            var bodyType = resourceType.Body.Type;
+
+            switch (bodyType)
+            {
+                case ObjectType bodyObjectType:
+                    bodyType = SetBicepResourceProperties(bodyObjectType, resourceType.ValidParentScopes, resourceType.TypeReference, flags);
+                    break;
+                default:
+                    // we exhaustively test deserialization of every resource type during CI, and this happens in a deterministic fashion,
+                    // so this exception should never occur in the released product
```

Comment:
This comment is inaccurate - for 3rd party providers, we have no way of exhaustively testing deserialization! I would just remove it.

We may want to create a work item to figure out how we better handle incorrectly serialized types. Not super high priority right now.

Reply:
Made a task here: https://msazure.visualstudio.com/One/_sprints/taskboard/Azure-ARM-Deployments/One/Germanium/CY24Q1/2Wk/2Wk08%20(Dec%2031%20-%20Jan%2013)?workitem=26429822